src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Projects/PackageReferenceProject.cs

```diff
@@ -186,7 +183,7 @@ internal async Task<ProjectPackages> GetInstalledAndTransitivePackagesAsync(bool
 
             // get transitive packages
             IEnumerable<PackageReference> calculatedTransitivePackages = Enumerable.Empty<PackageReference>();
-            if (includeTransitivePackages)
+            if (includeTransitivePackages || IsInstalledAndTransitiveComputationNeeded)
```

Comment:
This is the same condition as line 137 which had to be false for us to be in this else. Can you verify that we need to check both again and that the conditional logic is correct?